### ISSUE TITLE
macOS (Apple Silicon) entire machine crashes upon running project due to per-pixel transparency

### ISSUE THREAD

**Issue by Veradictus** [godotengine/godot#43951]
**Godot version:**
Godot 3.2.3
Godot 3.2.4 beta 3

**OS/device including version:**
MacBook Pro 2020 w/ Apple M1

**Issue description:**
Entire machine crashes when running a project in GoDot. The same behaviour is exhibited when trying to run an exported application.

**Comment by Veradictus** [comment]
Crash Details

```

panic(cpu 4 caller 0xfffffe002f9b67dc): Kernel data abort. at pc 0xfffffe002f81da5c, lr 0xfffffe0031343970 (saved state: 0xfffffe3fe05fb3d0)
	  x0: 0x000000000000034c  x1:  0xfffffe0032f6a3a0  x2:  0x0000000000000001  x3:  0xfffffe1679e89760
	  x4: 0xfffffe1679e89400  x5:  0x00000000000040d2  x6:  0x0000000000000000  x7:  0x0000000000000000
	  x8: 0x0000000000000001  x9:  0xffbdfe1ee53500a1  x10: 0x0000000000000009  x11: 0x000000000000011f
	  x12: 0xfacadf2c8d5df2e7 x13: 0xfffffe1679e88fc0  x14: 0x00000000000013b0  x15: 0x0000000000000001
	  x16: 0xfffffe002f81da58 x17: 0x00000000000013b0  x18: 0x0000000000000000  x19: 0x0000000000000000
	  x20: 0xfffffe1674a407a8 x21: 0x0000000000000000  x22: 0xfffffe2334c95c68  x23: 0x0000000000000003
	  x24: 0xfffffe2334c95b00 x25: 0x0000000000000001  x26: 0xfffffe002eeb47da  x27: 0xfffffe002eeb47f3
	  x28: 0xfffffe002eeb4802 fp:  0xfffffe3fe05fb790  lr:  0xfffffe0031343970  sp:  0xfffffe3fe05fb720
	  pc:  0xfffffe002f81da5c cpsr: 0x60401208         esr: 0x96000006          far: 0x000000000000034c

Debugger message: panic
Memory ID: 0x6
OS release type: User
OS version: 20B29
Kernel version: Darwin Kernel Version 20.1.0: Sat Oct 31 00:07:10 PDT 2020; root:xnu-7195.50.7~2/RELEASE_ARM64_T8101
Fileset Kernelcache UUID: 35A645C8D62619060673BEF751E6A7AA
Kernel UUID: D8523129-E878-38BA-A8E0-1D0E8D2470B7
iBoot version: iBoot-6723.50.2
secure boot?: YES
Paniclog version: 13
KernelCache slide: 0x0000000027498000
KernelCache base:  0xfffffe002e49c000
Kernel slide:      0x0000000028104000
Kernel text base:  0xfffffe002f108000
Kernel text exec base:  0xfffffe002f1d0000
mach_absolute_time: 0x407da9e25
Epoch Time:        sec       usec
  Boot    : 0x5fc3674d 0x00008a0b
  Sleep   : 0x00000000 0x00000000
  Wake    : 0x00000000 0x00000000
  Calendar: 0x5fc36a18 0x0005bd5c

CORE 0 recently retired instr at 0xfffffe002f33a4d0
CORE 1 recently retired instr at 0xfffffe002f33a4d0
CORE 2 recently retired instr at 0xfffffe002f33a4d0
CORE 3 recently retired instr at 0xfffffe002f33a4d0
CORE 4 recently retired instr at 0xfffffe002f33900c
CORE 5 recently retired instr at 0xfffffe002f33a4d4
CORE 6 recently retired instr at 0xfffffe002f33a4d4
CORE 7 recently retired instr at 0xfffffe002f33a4d4
Panicked task 0xfffffe16732898e0: 2517 pages, 10 threads: pid 1132: Godot
Panicked thread: 0xfffffe166fcb1308, backtrace: 0xfffffe3fe05fab40, tid: 16594
		  lr: 0xfffffe002f21df7c  fp: 0xfffffe3fe05fabb0
		  lr: 0xfffffe002f21dd48  fp: 0xfffffe3fe05fac20
		  lr: 0xfffffe002f33f7d0  fp: 0xfffffe3fe05fac40
		  lr: 0xfffffe002f3312fc  fp: 0xfffffe3fe05facf0
		  lr: 0xfffffe002f1d77e8  fp: 0xfffffe3fe05fad00
		  lr: 0xfffffe002f21d9d8  fp: 0xfffffe3fe05fb090
		  lr: 0xfffffe002f21d9d8  fp: 0xfffffe3fe05fb100
		  lr: 0xfffffe002f9b5dec  fp: 0xfffffe3fe05fb120
		  lr: 0xfffffe002f9b67dc  fp: 0xfffffe3fe05fb290
		  lr: 0xfffffe002f3331ac  fp: 0xfffffe3fe05fb300
		  lr: 0xfffffe002f331228  fp: 0xfffffe3fe05fb3b0
		  lr: 0xfffffe002f1d77e8  fp: 0xfffffe3fe05fb3c0
		  lr: 0xfffffe0031343970  fp: 0xfffffe3fe05fb790
		  lr: 0xfffffe0031343b10  fp: 0xfffffe3fe05fb7c0
		  lr: 0xfffffe00313428c8  fp: 0xfffffe3fe05fb810
		  lr: 0xfffffe0031342f68  fp: 0xfffffe3fe05fb840
		  lr: 0xfffffe003133f3f0  fp: 0xfffffe3fe05fb8c0
		  lr: 0xfffffe002f9266a4  fp: 0xfffffe3fe05fb920
		  lr: 0xfffffe002f933808  fp: 0xfffffe3fe05fbab0
		  lr: 0xfffffe002f30135c  fp: 0xfffffe3fe05fbbd0
		  lr: 0xfffffe002f2231dc  fp: 0xfffffe3fe05fbc40
		  lr: 0xfffffe002f1fbbf4  fp: 0xfffffe3fe05fbcc0
		  lr: 0xfffffe002f213dc8  fp: 0xfffffe3fe05fbd70
		  lr: 0xfffffe002f324bb0  fp: 0xfffffe3fe05fbe40
		  lr: 0xfffffe002f331370  fp: 0xfffffe3fe05fbef0
		  lr: 0xfffffe002f1d77e8  fp: 0xfffffe3fe05fbf00
      Kernel Extensions in backtrace:
         com.apple.iokit.IOGPUFamily(20.9)[F7B992A0-B500-3CAF-B8DC-C697F67934FC]@0xfffffe003133c000->0xfffffe003136bfff
            dependency: com.apple.driver.AppleMobileFileIntegrity(1.0.5)[E8AB2386-9256-3ACE-9CD9-44845CEA3E84]@0xfffffe0030480000->0xfffffe0030493fff
            dependency: com.apple.iokit.IOGraphicsFamily(585)[0767470E-7847-3BC4-842C-F54B5311ADE7]@0xfffffe003136c000->0xfffffe00313a3fff
            dependency: com.apple.iokit.IOPCIFamily(2.9)[B914AF0B-A3B4-3FC3-BED5-23D9700B0C1B]@0xfffffe00315e8000->0xfffffe0031603fff
            dependency: com.apple.iokit.IOReportFamily(47)[E9EB2287-9D5B-3323-AB8E-C8C0D2DD7096]@0xfffffe0031604000->0xfffffe0031607fff
            dependency: com.apple.iokit.IOSurface(289.3)[8AFE4AE4-DB5E-3ADC-B568-FE26A94E6F6F]@0xfffffe00316fc000->0xfffffe003171bfff

last started kext at 716971764: com.apple.filesystems.autofs	3.0 (addr 0xfffffe002f084000, size 16384)
loaded kexts:
com.apple.filesystems.autofs	3.0
com.apple.fileutil	20.036.15
com.apple.driver.AppleTopCaseHIDEventDriver	4000.27
com.apple.iokit.IOBluetoothSerialManager	8.0.1f5
com.apple.driver.AppleBiometricServices	1
com.apple.driver.BCMWLANFirmware4378.Hashstore	1
com.apple.driver.DiskImages.ReadWriteDiskImage	493.0.0
com.apple.driver.DiskImages.UDIFDiskImage	493.0.0
com.apple.driver.DiskImages.RAMBackingStore	493.0.0
com.apple.driver.DiskImages.FileBackingStore	493.0.0
com.apple.driver.CoreKDL	1
com.apple.driver.SEPHibernation	1
com.apple.filesystems.apfs	1677.50.1
com.apple.driver.AppleUSBDeviceNCM	5.0.0
com.apple.driver.AppleSmartBatteryManager	161.0.0
com.apple.driver.AppleThunderboltIP	4.0.3
com.apple.driver.AppleALSColorSensor	1.0.0d1
com.apple.driver.AppleFileSystemDriver	3.0.1
com.apple.driver.AppleAOPVoiceTrigger	11.5
com.apple.nke.l2tp	1.9
com.apple.filesystems.tmpfs	1
com.apple.IOTextEncryptionFamily	1.0.0
com.apple.filesystems.hfs.kext	556.41.1
com.apple.security.BootPolicy	1
com.apple.BootCache	40
com.apple.AppleFSCompression.AppleFSCompressionTypeZlib	1.0.0
com.apple.AppleFSCompression.AppleFSCompressionTypeDataless	1.0.0d1
com.apple.driver.ApplePMP	1
com.apple.driver.AppleSmartIO2	1
com.apple.driver.ApplePMPFirmware	1
com.apple.driver.AppleDPDisplayTCON	1
com.apple.AppleEmbeddedSimpleSPINORFlasher	1
com.apple.driver.AppleTAS5770LAmp	437.96
com.apple.driver.AppleSPMIPMU	1.0.1
com.apple.driver.AppleCS42L83Audio	437.96
com.apple.driver.AppleSummitLCD	1.0.0
com.apple.driver.AppleT8020SOCTuner	1
com.apple.driver.AppleT8103CLPCv3	1
com.apple.AGXG13G	172.20.9
com.apple.driver.AppleAVD	373
com.apple.driver.AppleAVE2	401.63.1
com.apple.driver.AppleJPEGDriver	4.6.0
com.apple.driver.AppleMobileDispH13G-DCP	140.0
com.apple.driver.usb.AppleUSBHostT8103	1
com.apple.driver.AudioDMAController-T8103	1.53
com.apple.driver.AppleS5L8960XNCO	1
com.apple.driver.AppleT8103PMGR	1
com.apple.driver.AppleS8000AES	1
com.apple.driver.AppleS8000DWI	1.0.0d1
com.apple.driver.AppleS5L8960XWatchDogTimer	1
com.apple.driver.AppleInterruptController	1.0.0d1
com.apple.driver.AppleBluetoothModule	1
com.apple.driver.AppleSamsungSerial	1.0.0d1
com.apple.driver.AppleBCMWLANBusInterfacePCIe	1
com.apple.driver.AppleT8020DART	1
com.apple.driver.AppleSynopsysMIPIDSI	1.0.0
com.apple.driver.AppleS5L8920XPWM	1.0.0d1
com.apple.driver.AppleS5L8940XI2C	1.0.0d2
com.apple.driver.AppleSPIMC	1
com.apple.driver.AppleM68Buttons	1.0.0d1
com.apple.driver.AppleT8101	1
com.apple.iokit.IOUserEthernet	1.0.1
com.apple.iokit.IOKitRegistryCompatibility	1
com.apple.iokit.EndpointSecurity	1
com.apple.driver.AppleDiskImages2	1
com.apple.AppleSystemPolicy	2.0.0
com.apple.nke.applicationfirewall	310
com.apple.kec.InvalidateHmac	1
com.apple.kext.triggers	1.0
com.apple.driver.AppleHSBluetoothDriver	4000.27
com.apple.driver.IOBluetoothHIDDriver	8.0.1f5
com.apple.driver.AppleActuatorDriver	4400.28
com.apple.driver.AppleMultitouchDriver	4400.28
com.apple.driver.AppleHIDKeyboard	222
com.apple.iokit.IOAVBFamily	900.12
com.apple.plugin.IOgPTPPlugin	900.11
com.apple.iokit.IOEthernetAVBController	1.1.0
com.apple.driver.AppleMesaSEPDriver	100.99
com.apple.iokit.IOBiometricFamily	1
com.apple.driver.DiskImages.KernelBacked	493.0.0
com.apple.driver.AppleSEPHDCPManager	1.0.1
com.apple.iokit.AppleSEPGenericTransfer	1
com.apple.driver.AppleBTM	1.0.1
com.apple.driver.AppleXsanScheme	3
com.apple.driver.IOBluetoothHostControllerPCIeTransport	8.0.1f5
com.apple.iokit.IOBluetoothHostControllerTransport	8.0.1f5
com.apple.driver.AppleConvergedIPCOLYBTControl	1
com.apple.driver.AppleConvergedPCI	1
com.apple.driver.AppleBluetoothDebug	1
com.apple.driver.usb.networking	5.0.0
com.apple.driver.AppleThunderboltPCIDownAdapter	4.1.1
com.apple.driver.AppleThunderboltDPInAdapter	8.1.4
com.apple.driver.AppleThunderboltDPAdapterFamily	8.1.4
com.apple.driver.AppleThunderboltUSBDownAdapter	1.0.4
com.apple.driver.AppleAOPAudio	14.32
com.apple.nke.ppp	1.9
com.apple.AGXFirmwareKextG13GRTBuddy	172.20.9
com.apple.AGXFirmwareKextRTBuddy64	172.20.9
com.apple.driver.AppleBSDKextStarter	3
com.apple.filesystems.hfs.encodings.kext	1
com.apple.driver.AppleDCPDPTXProxy	1.0.0
com.apple.driver.DCPDPFamilyProxy	1
com.apple.iokit.IONVMeFamily	2.1.0
com.apple.driver.AppleSPU	1
com.apple.driver.AppleHIDTransportSPI	4400.35
com.apple.driver.AppleHIDTransport	4400.35
com.apple.driver.AppleInputDeviceSupport	4400.35
com.apple.driver.AppleHPM	3.4.4
com.apple.driver.AppleDiagnosticDataAccessReadOnly	1.0.0
com.apple.driver.AppleNANDConfigAccess	1.0.0
com.apple.driver.AppleStockholmControl	1.0.0
com.apple.driver.AppleDialogPMU	1.0.1
com.apple.driver.AppleCSEmbeddedAudio	437.96
com.apple.driver.AppleEmbeddedAudio	437.96
com.apple.iokit.AppleARMIISAudio	80.34
com.apple.driver.ApplePassthroughPPM	3.0
com.apple.iokit.IOGPUFamily	20.9
com.apple.driver.AppleH11ANEInterface	4.51.0
com.apple.driver.AppleH13CameraInterface	2.67.2
com.apple.driver.AppleH10PearlCameraInterface	14.63.0
com.apple.driver.DCPAVFamilyProxy	1
com.apple.iokit.IOMobileGraphicsFamily-DCP	343.0.0
com.apple.driver.AppleDCP	1
com.apple.driver.AppleFirmwareKit	1
com.apple.driver.AppleSPMI	1.0.1
com.apple.driver.AppleUSBXDCIARM	1.0
com.apple.driver.AppleUSBXDCI	1.0
com.apple.iokit.IOUSBDeviceFamily	2.0.0
com.apple.driver.usb.AppleUSBXHCIARM	1
com.apple.driver.usb.AppleUSBXHCI	1.2
com.apple.driver.AppleEmbeddedUSBHost	1
com.apple.driver.usb.AppleUSBHub	1.2
com.apple.driver.usb.AppleUSBHostCompositeDevice	1.2
com.apple.driver.AppleT8103TypeCPhy	1
com.apple.driver.AppleSART	1
com.apple.driver.ApplePMGR	1
com.apple.driver.watchdog	1
com.apple.driver.usb.AppleUSBHostPacketFilter	1.0
com.apple.driver.AppleDisplayCrossbar	1.0.0
com.apple.iokit.IODisplayPortFamily	1.0.0
com.apple.driver.AppleTypeCPhy	1
com.apple.driver.AppleThunderboltNHI	7.2.8
com.apple.driver.AppleT8103PCIeC	1
com.apple.iokit.IOThunderboltFamily	9.3.2
com.apple.driver.ApplePIODMA	1
com.apple.driver.AppleT8103PCIe	1
com.apple.driver.AppleBluetoothDebugService	1
com.apple.driver.AppleBCMWLANCore	1.0.0
com.apple.iokit.IO80211FamilyV2	1200.12.2b1
com.apple.driver.IOImageLoader	1.0.0
com.apple.driver.AppleMobileDispH13G-DFR	140.0
com.apple.iokit.IOMobileGraphicsFamily	343.0.0
com.apple.driver.AppleMCA2-T8103	510.72
com.apple.driver.AppleGPIOICController	1.0.2
com.apple.driver.AppleMobileApNonce	1
com.apple.driver.AppleFireStormErrorHandler	1
com.apple.driver.AppleMultiFunctionManager	1
com.apple.driver.corecapture	1.0.4
com.apple.driver.AppleEmbeddedPCIE	1
com.apple.iokit.IOTimeSyncFamily	900.11
com.apple.driver.DiskImages	493.0.0
com.apple.iokit.IOGraphicsFamily	585
com.apple.iokit.IOBluetoothFamily	8.0.1f5
com.apple.iokit.IOBluetoothPacketLogger	8.0.1f5
com.apple.driver.FairPlayIOKit	68.5.0
com.apple.iokit.CoreAnalyticsFamily	1
com.apple.driver.AppleSSE	1.0
com.apple.driver.AppleSEPKeyStore	2
com.apple.driver.AppleUSBTDM	511.40.9
com.apple.iokit.IOUSBMassStorageDriver	184.40.6
com.apple.iokit.IOPCIFamily	2.9
com.apple.iokit.IOSCSIBlockCommandsDevice	436.40.6
com.apple.iokit.IOSCSIArchitectureModelFamily	436.40.6
com.apple.driver.AppleIPAppender	1.0
com.apple.driver.AppleFDEKeyStore	28.30
com.apple.driver.AppleEffaceableStorage	1.0
com.apple.driver.AppleCredentialManager	1.0
com.apple.driver.KernelRelayHost	1
com.apple.iokit.IOUSBHostFamily	1.2
com.apple.driver.AppleUSBHostMergeProperties	1.2
com.apple.driver.usb.AppleUSBCommon	1.0
com.apple.driver.AppleSMC	3.1.9
com.apple.driver.RTBuddy	1.0.0
com.apple.driver.AppleEmbeddedTempSensor	1.0.0
com.apple.driver.AppleARMPMU	1.0
com.apple.iokit.IOAccessoryManager	1.0.0
com.apple.driver.AppleOnboardSerial	1.0
com.apple.iokit.IOSkywalkFamily	1
com.apple.driver.mDNSOffloadUserClient	1.0.1b8
com.apple.iokit.IONetworkingFamily	3.4
com.apple.iokit.IOSerialFamily	11
com.apple.driver.AppleSEPManager	1.0.1
com.apple.driver.AppleA7IOP	1.0.2
com.apple.driver.IOSlaveProcessor	1
com.apple.driver.AppleBiometricSensor	2
com.apple.iokit.IOHIDFamily	2.0.0
com.apple.AUC	1.0
com.apple.iokit.IOAVFamily	1.0.0
com.apple.iokit.IOHDCPFamily	1.0.0
com.apple.iokit.IOCECFamily	1
com.apple.iokit.IOAudio2Family	1.0
com.apple.driver.AppleEmbeddedAudioLibs	1.17
com.apple.driver.AppleFirmwareUpdateKext	1
com.apple.driver.AppleM2ScalerCSCDriver	265.0.0
com.apple.iokit.IOSurface	289.3
com.apple.driver.IODARTFamily	1
com.apple.security.quarantine	4
com.apple.security.sandbox	300.0
com.apple.kext.AppleMatch	1.0.0d1
com.apple.driver.AppleMobileFileIntegrity	1.0.5
com.apple.kext.CoreTrust	1
com.apple.security.AppleImage4	3.0.0
com.apple.iokit.IOCryptoAcceleratorFamily	1.0.1
com.apple.driver.AppleARMPlatform	1.0.2
com.apple.iokit.IOStorageFamily	2.1
com.apple.iokit.IOSlowAdaptiveClockingFamily	1.0.0
com.apple.iokit.IOReportFamily	47
com.apple.kec.pthread	1
com.apple.kec.corecrypto	1.0
com.apple.kec.Libm	1



** Stackshot Succeeded ** Bytes Traced 283076 (Uncompressed 724896) **
```

**Comment by Calinou** [comment]
cc @bruvzg

**Comment by Veradictus** [comment]
Update, this bug is caused by ONE of the following variables in the GoDot Configuration. I haven't had time to test which one of these causes the crash since I don't really enjoy crashing my computer repeatedly.

```
window/per_pixel_transparency/allowed=true
window/per_pixel_transparency/enabled=true
window/stretch/aspect="keep"

quality/driver/fallback_to_gles2=true
quality/2d/use_nvidia_rect_flicker_workaround=true```

**Comment by Calinou** [comment]
Confirmed here: https://godotengine.org/qa/89458/per-pixel-transparency-lets-my-mac-crash

**Comment by Veradictus** [comment]
@Calinou I've been using 3.2.4beta3 for a while now. It still causes that issue if I enable per-pixel-transparency.

**Comment by eleanor-em** [comment]
Still have the same issue with the latest beta version.

**Comment by photex** [comment]
Can confirm that this issue is still present in 3.3.2.stable. Mac mini (apple silicon) MacOS 11.3.

**Comment by chrisg32** [comment]
I am also experiencing this on the M1 with 3.3.2 stable. All other variables constant the per pixel transparency is what causes the crash.

**Comment by akien-mga** [comment]
Fixed by #55464.
